File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
    int left = 0; 
    int right = nums.length-1; 
    
    while(left < right){ 
        int middle = (left + right)/2;
        if(middle == left){
            return -1;
        }
        if(nums[middle] < target){ 
            left = middle;
        } else if(nums[middle] > target){ 
            right = middle;
        } else { 
            return middle;
        }
    }

    return -1;



    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row = 0 

        while(row <= len(matrix)):
            left = 0 
            right = len(matrix[0])
            nums = matrix[row]

            if(target >= nums[left] and target <= nums[right]): 
                while(left <= right): 
                    mid = left + (right - left)//2 

                    if(nums[mid] == target): 
                        return True
                    elif(nums[mid] > target): 
                        left = mid + 1
                    elif(nums[mid] < target): 
                        right = mid - 1
                    else: 
                        return False
            else: 
                row += 1


        return False